nibabel/cifti2/tests/test_cifti2.py

```diff
@@ -27,244 +27,287 @@ def compare_xml_leaf(str1, str2):
-    assert_raises(ValueError, _float_01, -0.1)
-    assert_raises(ValueError, _float_01, 2)
-    assert_raises(ValueError, _float_01, -1)
-    assert_raises(ValueError, _float_01, 'foo')
+    assert _float_01(0) == 0
+    assert _float_01(1) == 1
+    assert _float_01('0') == 0
+    assert _float_01('0.2') == 0.2
+    with pytest.raises(ValueError):
+        _float_01(1.1)
+    with pytest.raises(ValueError):
+        _float_01(-0.1)
+    with pytest.raises(ValueError):
+        _float_01(2)
+    with pytest.raises(ValueError):
+        _float_01(-1)
+    with pytest.raises(ValueError):
+        _float_01('foo')
 
 
 def test_cifti2_labeltable():
     lt = ci.Cifti2LabelTable()
-    assert_equal(len(lt), 0)
-    assert_raises(ci.Cifti2HeaderError, lt.to_xml)
-    assert_raises(ci.Cifti2HeaderError, lt._to_xml_element)
+    assert len(lt) == 0
+    with pytest.raises(ci.Cifti2HeaderError):
+        lt.to_xml()
+    with pytest.raises(ci.Cifti2HeaderError):
+        lt._to_xml_element()
+    
 
     label = ci.Cifti2Label(label='Test', key=0)
     lt[0] = label
-    assert_equal(len(lt), 1)
-    assert_equal(dict(lt), {label.key: label})
+    assert len(lt) == 1
+    assert dict(lt) == {label.key: label}
 
     lt.clear()
     lt.append(label)
-    assert_equal(len(lt), 1)
-    assert_equal(dict(lt), {label.key: label})
+    assert len(lt) == 1
+    assert dict(lt) == {label.key: label}
 
     lt.clear()
     test_tuple = (label.label, label.red, label.green, label.blue, label.alpha)
     lt[label.key] = test_tuple
-    assert_equal(len(lt), 1)
+    assert len(lt) == 1
     v = lt[label.key]
-    assert_equal(
-        (v.label, v.red, v.green, v.blue, v.alpha),
-        test_tuple
-    )
+    assert (v.label, v.red, v.green, v.blue, v.alpha) == test_tuple
+
+    with pytest.raises(ValueError):
+        lt[1] = label
+
+    with pytest.raises(ValueError):
+        lt[0] = test_tuple[:-1]
+    
+    with pytest.raises(ValueError):
+        lt[0] = ('foo', 1.1, 0, 0, 1)   
+
+    with pytest.raises(ValueError):
+        lt[0] = ('foo', 1.0, -1, 0, 1)
+
+    with pytest.raises(ValueError):
+        lt[0] = ('foo', 1.0, 0, -0.1, 1)
 
-    assert_raises(ValueError, lt.__setitem__, 1, label)
-    assert_raises(ValueError, lt.__setitem__, 0, test_tuple[:-1])
-    assert_raises(ValueError, lt.__setitem__, 0, ('foo', 1.1, 0, 0, 1))
-    assert_raises(ValueError, lt.__setitem__, 0, ('foo', 1.0, -1, 0, 1))
-    assert_raises(ValueError, lt.__setitem__, 0, ('foo', 1.0, 0, -0.1, 1))
 
 
 def test_cifti2_label():
     lb = ci.Cifti2Label()
     lb.label = 'Test'
     lb.key = 0
-    assert_equal(lb.rgba, (0, 0, 0, 0))
-    assert_true(compare_xml_leaf(
-        lb.to_xml().decode('utf-8'),
-        "<Label Key='0' Red='0' Green='0' Blue='0' Alpha='0'>Test</Label>"
-    ))
+    assert lb.rgba == (0, 0, 0, 0)
+    assert compare_xml_leaf(lb.to_xml().decode('utf-8'), "<Label Key='0' Red='0' Green='0' Blue='0' Alpha='0'>Test</Label>")
```

Comment:
I didn't check the character count, but if its going off the side of GitHub, it's probably too long:

```suggestion
    assert compare_xml_leaf(lb.to_xml().decode('utf-8'),
                            "<Label Key='0' Red='0' Green='0' Blue='0' Alpha='0'>Test</Label>")
```